finetuner/tuner/callback/best_model_checkpoint.py

```diff
@@ -0,0 +1,140 @@
+import os
+from typing import TYPE_CHECKING, Optional
+
+import numpy as np
+from jina.logging.logger import JinaLogger
+
+from finetuner.helper import get_framework
+from .base import BaseCallback
+
+
+if TYPE_CHECKING:
+    from ..base import BaseTuner
+
+
+class BestModelCheckpoint(BaseCallback):
+    """
+    Callback to save model at every epoch or the best model across all epochs
+    `BestModelCheckpoint` is used in conjunction with training
+    using `finetuner.fit()`
+
+    An option this callback provides include:
+    - Definition of 'best'; which quantity to monitor and whether it should be
+        maximized or minimized.
+    """
+
+    def __init__(
+        self,
+        save_dir: str = None,
+        monitor: str = 'val_loss',
+        mode: Optional[str] = None,
```

Comment:
Can we have this `auto` as default, and not `Optional`? Better explicit than implicit